source/grid/grid_generator.cc

```diff
@@ -7090,6 +7090,273 @@ namespace GridGenerator
+              }
+            else
+              {
+                Assert(false, ExcNotImplemented());
+              }
+          };
+
+        const auto current_material_id = cell.material_id();
+
+        // create cells one by one
+        if (dim == 2)
+          {
+            add_cell(dim, {0, 6, 4}, current_material_id);
+            add_cell(dim, {8, 4, 6}, current_material_id);
+            add_cell(dim, {8, 6, 5}, current_material_id);
+            add_cell(dim, {1, 5, 6}, current_material_id);
+            add_cell(dim, {2, 4, 7}, current_material_id);
+            add_cell(dim, {8, 7, 4}, current_material_id);
+            add_cell(dim, {8, 5, 7}, current_material_id);
+            add_cell(dim, {3, 7, 5}, current_material_id);
+          }
+        else if (dim == 3)
+          {
+            // large tets parallel x axis
+            add_cell(dim, {0, 1, 12, 10}, current_material_id);
+            add_cell(dim, {2, 3, 11, 12}, current_material_id);
+            add_cell(dim, {7, 6, 11, 13}, current_material_id);
+            add_cell(dim, {5, 4, 13, 10}, current_material_id);
+
+            // large tets parallel y axis
+            add_cell(dim, {0, 2, 8, 12}, current_material_id);
+            add_cell(dim, {4, 6, 13, 8}, current_material_id);
+            add_cell(dim, {5, 13, 7, 9}, current_material_id);
+            add_cell(dim, {1, 9, 3, 12}, current_material_id);
+
+            // large tets parallel z axis
+            add_cell(dim, {0, 8, 4, 10}, current_material_id);
+            add_cell(dim, {1, 5, 9, 10}, current_material_id);
+            add_cell(dim, {3, 7, 11, 9}, current_material_id);
+            add_cell(dim, {2, 6, 8, 11}, current_material_id);
+
+            // large tets in middle octaeder
+            add_cell(dim, {12, 13, 10, 9}, current_material_id);
+            add_cell(dim, {12, 13, 9, 11}, current_material_id);
+            add_cell(dim, {12, 13, 11, 8}, current_material_id);
+            add_cell(dim, {12, 13, 8, 10}, current_material_id);
+
+            // small tets in corners
+            add_cell(dim, {13, 8, 10, 4}, current_material_id);
+            add_cell(dim, {13, 10, 9, 5}, current_material_id);
+            add_cell(dim, {13, 9, 11, 7}, current_material_id);
+            add_cell(dim, {13, 11, 8, 6}, current_material_id);
+
+            add_cell(dim, {10, 12, 9, 1}, current_material_id);
+            add_cell(dim, {9, 12, 11, 3}, current_material_id);
+            add_cell(dim, {11, 12, 8, 2}, current_material_id);
+            add_cell(dim, {8, 12, 10, 0}, current_material_id);
+          }
+        else
+          {
+            Assert(false, ExcNotImplemented());
+          }
+
+        // Set up sub-cell data.
+        for (const auto f : cell.face_indices())
+          {
+            const auto current_boundary_id = cell.face(f)->boundary_id();
+            if (current_boundary_id == numbers::internal_face_boundary_id)
+              continue;
```

Comment:
It is easier to read if you write this as
```
  for (const auto f : ...)
    if (cell.face(f)->boundary_id() != numbers::internal_face_boundary_id)
      {
         ...
```
This way it is clear at the top of the loop which faces will actually be processed.